pkg/parser/reader.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package parser
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/afero"
+)
+
+const (
+	yamlExt      = ".yaml"
+	yamlExtShort = ".yml"
+)
+
+// FsReadCloser implements io.ReadCloser for an Afero filesystem.
+type FsReadCloser struct {
+	fs       afero.Fs
+	dir      string
+	files    []os.FileInfo
+	index    int
+	position int
+}
+
+// NewFsReadCloser returns an FsReadCloser that implements io.ReadCloser.
+func NewFsReadCloser(fs afero.Fs, dir string) (*FsReadCloser, error) {
+	files, err := afero.ReadDir(fs, dir)
+	return &FsReadCloser{
+		fs:       fs,
+		dir:      dir,
+		files:    files,
+		index:    0,
+		position: 0,
+	}, err
+}
+
+func (r *FsReadCloser) Read(p []byte) (n int, err error) {
```

Comment:
Is this guy intended to turn the content of a directory into a YAML stream? i.e. Is this what the unpack process would call to parse the content of an OCI image? If so, I think it will need to find files that are in nested directories, and also accept a list of paths to ignore. Perhaps https://godoc.org/github.com/spf13/afero#Walk would be more appropriate?

Reply:
Good point! I think at `NewFsReadCloser(fs afero.Fs, dir string)` we could just pass in an `afero.WalkFn` and call `Walk` instead of `ReadDir`. That `WalkFn` could do things like filter out ignored paths. How does that sound?

Reply:
@negz I did a variation of this in most recent push, could you take a look and see what you think?